File: kotlin/src/main/kotlin/com/cardinal/utils/CreateSDK.kt
package com.cardinal.utils

import com.icure.cardinal.sdk.CardinalSdk
import com.icure.cardinal.sdk.auth.UsernamePassword
import com.icure.cardinal.sdk.options.AuthenticationMethod
import com.icure.cardinal.sdk.storage.impl.FileStorageFacade

private const val CARDINAL_URL = "https://api.icure.cloud"


suspend fun createSdk(username: String, password: String): CardinalSdk = CardinalSdk.initialize(
	applicationId = null,
	baseUrl = CARDINAL_URL,
	authenticationMethod = AuthenticationMethod.UsingCredentials(
		UsernamePassword(username, password)
	),
	baseStorage = FileStorageFacade("./scratch/storage")
)

File: python/src/utils/create_sdk.py
from cardinal_sdk import CardinalSdk
from cardinal_sdk.authentication.AuthenticationMethod import UsernamePassword
from cardinal_sdk.storage.StorageFacadeOptions import FileSystemStorage

CARDINAL_URL = "https://api.icure.cloud"


def create_sdk(username: str, password: str) -> CardinalSdk:
	return CardinalSdk(
		application_id=None,
		baseurl=CARDINAL_URL,
		authentication_method=UsernamePassword(username, password),
		storage_facade=FileSystemStorage("../scratch/storage")
	)
